Automattic/sensei
# Add importer UI data manager

### Changes proposed in this Pull Request

* To better be able to manage state across the importer UI, I followed @renatho's lead from his work in the setup wizard (#3157). 
* This introduces a central data store for importer state. 
* Refactors the navigation to restore from the fetched state.
* Refactors the upload step to use the central state. It also restores the uploader's state when refreshing the page with an import job partially set up. In order to do this in a way that was testable, I had to separate the component from the container that injected the selectors and actions into the props.
* In addition, this renames `.jsx` to `.js` and adds `sensei-wc-components` style dependency to the importer styles.

- Note: I am pretty new to using WP Data so would really appreciate any feedback on how I implemented this.

### Testing instructions

* Build source.
* Go to Sensei LMS > Import.
* Upload a file. Notice you can continue past that step after one file has successfully been uploaded.
* Refresh the page. Notice the file state has been restored.
* Continue. You should get to the progress /`Import` placeholder page.
* Refresh page. You should be brought back to that page.
